pandas/core/indexes/base.py

```diff
@@ -567,6 +567,14 @@ def __new__(
                 # Ensure we get 1-D array of tuples instead of 2D array.
                 data = com.asarray_tuplesafe(data, dtype=_dtype_obj)
 
+            # 60925 should raise when one of index's items is a list and others are not
+            if any(isinstance(el, list) for el in data) and not all(
+                isinstance(el, list) for el in data
+            ):
```

Comment:
No non-hashable value should occur here. Instead, can you do this check on `arr` below, and only in the case it's (a) a NumPy array (not an ExtensionArray) and (b) of object dtype. Use `pandas.core.dtypes.inference.is_hashable`.